src/js/common/schemaform/FieldTemplate.jsx

```diff
@@ -56,21 +58,46 @@ export default function FieldTemplate(props) {
     'usa-input-error form-error-date': isDateField && hasErrors
   });
 
-  return (schema.type === 'object'
-    || schema.type === 'array'
-    || (schema.type === 'boolean' && !uiSchema['ui:widget'])
-  ) && !showFieldLabel
-    ? children
-    : (<div className={containerClassNames}>
-      <div className={errorClass}>
-        {!hideLabelText && <label className={labelClassNames} htmlFor={id}>{label}{requiredSpan}</label>}
-        {textDescription && <p>{textDescription}</p>}
-        {DescriptionField && <DescriptionField options={uiSchema['ui:options']}/>}
-        {!textDescription && !DescriptionField && description}
-        {errorSpan}
-        {<div className={inputWrapperClassNames}>{children}</div>}
-        {help}
-      </div>
+  const noWrapperContent = !showFieldLabel &&
+    (
+      schema.type === 'object' ||
+      schema.type === 'array' ||
+      (schema.type === 'boolean' && !uiSchema['ui:widget'])
+    );
+
+  if (noWrapperContent) {
+    return children;
+  }
+
+  const useFieldsetLegend = (isFieldGroup || showFieldLabel) && !useLabel;
+
+  const labelElement = useFieldsetLegend
+    ? <legend className={labelClassNames}>{label}{requiredSpan}</legend>
+    : <label className={labelClassNames} htmlFor={id}>{label}{requiredSpan}</label>;
+
+  const content = (
+    <div className={errorClass}>
+      {!hideLabelText && labelElement}
+      {textDescription && <p>{textDescription}</p>}
+      {DescriptionField && <DescriptionField options={uiSchema['ui:options']}/>}
+      {!textDescription && !DescriptionField && description}
+      {errorSpan}
+      {<div className={inputWrapperClassNames}>{children}</div>}
+      {help}
     </div>
+  );
+
+  if (useFieldsetLegend) {
+    return (
+      <fieldset className={containerClassNames}>
+        {content}
+      </fieldset>
     );
+  }
+
+  return (
+    <div className={containerClassNames}>
```

Comment:
Idle curiosity: What would happen if we always used a `fieldset`? Is this separate return so we don't use a `label` in a `fieldset` (because of reasons I don't know about)?

Reply:
It feels semantically wrong to always use `fieldset`, since it's meant to group related elements. If it's just one field, it's not really a group. I don't think it would break anything, though.